### PR TITLE
Update status when deleted containers

### LINKED ISSUE #88406: Deleting pod: exited containers marked as running until last one exits
**What happened**:
I have two container running. When deleting the pod containing them, both containers remain in state _running_ until the last one has exited.

**What you expected to happen**:
The first container to shutdown gracefully should be marked as completed as soon as it exits.

**How to reproduce it (as minimally and precisely as possible)**:
Run the following pod, and delete it. The one which exits after 10 seconds continues to claim it is running, even though it has exited.

```yaml
apiVersion: v1
kind: Pod
metadata:
  name: test-termination
  namespace: default
spec:
  restartPolicy: Never
  terminationGracePeriodSeconds: 60
  containers:
    - name: short-terminate
      image: "alpine:3.9.5"
      resources:
        limits:
          cpu: 0.1
          memory: "50Mi"
        requests:
          cpu: 0.1
          memory: "50Mi"
      command: ["/bin/sh",  "-c"]
      args: ["
        echo started;
        trap 'echo SIGTERM received; sleep 10; echo done; exit' SIGTERM;
        while true;
        do
          sleep 2;
        done"]
    - name: long-terminate
      image: "alpine:3.9.5"
      resources:
        limits:
          cpu: 0.1
          memory: "50Mi"
        requests:
          cpu: 0.1
          memory: "50Mi"
      command: ["/bin/sh",  "-c"]
      args: ["
        echo started;
        trap 'echo SIGTERM received, wait 120s; sleep 120; echo done; exit' SIGTERM;
        while true;
        do
          sleep 2;
        done"]
```

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.3", GitCommit:"b3cbbae08ec52a7fc73d334838e18d17e8512749", GitTreeState:"clean", BuildDate:"2019-11-14T04:24:29Z", GoVersion:"go1.12.13", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"15+", GitVersion:"v1.15.9-gke.8", GitCommit:"a9973cbb2722793e2ea08d20880633ca61d3e669", GitTreeState:"clean", BuildDate:"2020-02-07T00:50:57Z", GoVersion:"go1.12.12b4", Compiler:"gc", Platform:"linux/amd64"}
```
- Cloud provider or hardware configuration: GCP managed kubernetes cluster

### PR DESCRIPTION
**Pull request by wzshiming**
#### What type of PR is this?
/kind feature
/sig node

#### What this PR does / why we need it:

Update status when deleted containers. Currently, the status is updated only after all containers are stopped, which is unintuitive behavior

#### Which issue(s) this PR fixes:

Fixes #88406

#### Does this PR introduce a user-facing change?

```release-note
Update status when deleted containers
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
NONE
```

### CODE REVIEW

**Comment by matthyx** [comment]
@wzshiming could you add some unit tests?

File: pkg/kubelet/kuberuntime/kuberuntime_manager.go [hunk 1]
```diff
@@ -951,14 +958,23 @@ func (m *kubeGenericRuntimeManager) killPodWithSyncResult(pod *v1.Pod, runningPo
...
-			klog.ErrorS(nil, "Failed to stop sandbox", "podSandboxID", podSandbox.ID)
-		}
+		go func(podSandbox *kubecontainer.Container) {
+			defer wg.Done()
+			if err := m.runtimeService.StopPodSandbox(podSandbox.ID.ID); err != nil {
+				killSandboxResult.Fail(kubecontainer.ErrKillPodSandbox, err.Error())
+				klog.ErrorS(nil, "Failed to stop sandbox", "podSandboxID", podSandbox.ID)
+			} else if m.statusManager != nil {
+				m.statusManager.SetContainerTerminated(pod.UID, podSandbox.ID)
+				klog.Info("Stop sandbox", "podSandboxID", podSandbox.ID)
```
Comment (matthyx) [inline pkg/kubelet/kuberuntime/kuberuntime_manager.go:973]: can you use structured logging here? https://kubernetes.io/blog/2020/09/04/kubernetes-1-19-introducing-structured-logs/

Comment (wzshiming) [reply pkg/kubelet/kuberuntime/kuberuntime_manager.go]: Thanks, updated.

**Comment by ehashman** [comment]
/priority important-longterm
/triage accepted

PTAL at the test failures.

**Review by matthyx — APPROVED** [review]

**Comment by wzshiming** [comment]
Friendly ping @mrunalp

**Comment by wzshiming** [comment]
hi @mrunalp could you have time to take a look

**Comment by mrunalp** [comment]
@wzshiming should sync with with @smarterclayton. He has a PR open https://github.com/kubernetes/kubernetes/pull/102344/files

**Comment by matthyx** [comment]
> @wzshiming should sync with with @smarterclayton. He has a PR open https://github.com/kubernetes/kubernetes/pull/102344/files

@wzshiming did you check with @smarterclayton ?

**Comment by dims** [comment]
Is this PR still needed, please rebase if so (or we can close it?)

**Comment by dims** [comment]
This PR is older than 4 weeks and needs as rebase.

Please rebase the PR against latest master and reopen if still needed.

/close

**Pull request closed without merging**